src/sentry/static/sentry/app/components/issueDiff/index.tsx

```diff
@@ -0,0 +1,190 @@
+import {css} from '@emotion/core';
+import React from 'react';
+import isPropValid from '@emotion/is-prop-valid';
+import styled from '@emotion/styled';
+
+import {Client} from 'app/api';
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {t} from 'app/locale';
+import withApi from 'app/utils/withApi';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import getStacktraceBody from 'app/utils/getStacktraceBody';
+import SplitDiff from 'app/components/splitDiff';
+import ButtonBar from 'app/components/buttonBar';
+import Button from 'app/components/button';
+import space from 'app/styles/space';
+import {Project} from 'app/types';
+
+import renderGroupingInfo from './groupingDiff';
+
+type Props = {
+  api: Client;
+  baseIssueId: string;
+  targetIssueId: string;
+
+  orgId: string;
+  project: Project;
+
+  baseEventId?: string;
+  targetEventId?: string;
+  className?: string;
+};
+
+type State = {
+  loading: boolean;
+  groupingDiff: boolean;
+  baseEvent: Array<string>;
+  targetEvent: Array<string>;
+  SplitDiffAsync?: typeof SplitDiff;
+};
+
+class IssueDiff extends React.Component<Props, State> {
+  static defaultProps = {
+    baseEventId: 'latest',
+    targetEventId: 'latest',
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      loading: true,
+      groupingDiff: false,
+      baseEvent: [],
+      targetEvent: [],
+
+      // `SplitDiffAsync` is an async-loaded component
+      // This will eventually contain a reference to the exported component from `./splitDiff`
+      SplitDiffAsync: undefined,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+    const {baseIssueId, targetIssueId, baseEventId, targetEventId} = this.props;
+
+    // Fetch component and event data
+    Promise.all([
+      import(/* webpackChunkName: "splitDiff" */ '../splitDiff'),
+      this.fetchEventData(baseIssueId, baseEventId || 'latest'),
+      this.fetchEventData(targetIssueId, targetEventId || 'latest'),
```

Comment:
these values will never be empty as we set them in the defaultProps.

```suggestion
      this.fetchEventData(baseIssueId, baseEventId),
      this.fetchEventData(targetIssueId, targetEventId),
```

Reply:
I think it is possible that they are empty if null is passed in explicitly. I did not want to remove this as the caller is JSX and I can't rely on typesafety just yet.